### PR TITLE
Update GOOGLE_RTC_AUDIO_PROCESSING for Zephyr

### PR DESCRIPTION
**Pull request by andyross**
Port over the existing integration for now.  This comes from an era where SOF had no C++ support and this module was effectively rolling its own.  Zephyr has a theoretically robust and featureful C++ layer, though it isn't exposed for Cadence toolchains yet (and until yesterday was actually plain broken for C++98 compilers -- you need Zephyr main to get this to build right now, the SOF snapshot is still behind).

Consider this a stepping stone to a feature that needs to land upstream.  Useful to have in a PR for reference, maybe not for merging right now.

Note that I discovered with horror that this crossed with the MTL switchover to xt-clang.  That configuration hasn't yet been tested, may not actually link, and will likely require two separate C++ integration layers upstream.

### CODE REVIEW

**Review by cujomalainey — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

File: zephyr/lib/alloc.c [hunk 1]
```diff
@@ -360,4 +360,15 @@ static int heap_init(const struct device *unused)
...
+/* This is a weak stub for the Cadence libc's allocator (which is just
+ * a newlib build).  It's traditionally been provided like this in SOF
+ * for the benefit of C++ code where the standard library needs to
+ * link to a working malloc() even if it will never call it.
+ */
+struct _reent;
+__weak void *_sbrk_r(struct _reent *ptr, ptrdiff_t incr)
+{
+	k_panic();
+}
```
Comment (marc-hb) [inline zephyr/lib/alloc.c:372]: https://github.com/thesofproject/sof/actions/runs/4317869629/jobs/7535474369
```
/zep_workspace/sof/zephyr/lib/alloc.c: In function '_sbrk_r':
/zep_workspace/sof/zephyr/lib/alloc.c:372:1: error: control reaches end of non-void function [-Werror=return-type]
  372 | }
      | ^
cc1: all warnings being treated as errors
```

**Review by marc-hb — CHANGES_REQUESTED** [review]
> Note that I discovered with horror that this crossed with the MTL switchover to xt-clang. That configuration hasn't yet been tested, may not actually link, and will likely require two separate C++ integration layers upstream.

Yes let's please not merge this until it's rebased on top of
- #7027

Or you could just drop the xtensa-build-zephyr.py commit from this PR to speed it up; it seems unrelated.

**Comment by kv2019i** [comment]
@andyross https://github.com/thesofproject/sof/pull/7027 now merged, please rebase.

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Comment by lgirdwood** [comment]
@andyross ping - needs a rebase

**Comment by cujomalainey** [comment]
Possibly won't happen for a couple weeks, we are quite busy on this end

**Comment by cujomalainey** [comment]
Ping to @andyross to follow up on this

**Comment by andyross** [comment]
(Rebased on mtl-005-drop-stable for ease of distribution.  Zephyr-side C++ integration still to come)

**Comment by lgirdwood** [comment]
> (Rebased on mtl-005-drop-stable for ease of distribution. Zephyr-side C++ integration still to come)

Needs to be rebased on main sinec this PR is for main.

**Comment by cujomalainey** [comment]
Looks like the rebase wasn't clean and pulled in a bunch of stuff

**Comment by abonislawski** [comment]
@andyross should we finally merge this to main & 007? As far as I know these commits are cherry-picked for AEC build

**Pull request closed without merging**